pds4indextools/pds4_create_collection_index.py

```diff
@@ -0,0 +1,106 @@
+"""
+PDS4 Collection Product Creation Tool
+
+This script scrapes label files within a collection for their identifying information,
+and creates a collection product file containing the LIDVIDs and the Member Status of the
+label files. This tool can also generate the collection product elsewhere within the
+user's directory, for use during the creation of collections. For the full instructions
+on how to use the tool, see:
+
+python pds4_create_collection_product.py --help
+"""
+
+import argparse
+import csv
+from lxml import etree
+from pathlib import Path
+import sys
+
+
+def main():
+
+    # Parse command line arguments
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('collectionpath', type=str, metavar='COLLECTION_PATH',
+                        help='The path to the collection product')
+    parser.add_argument('--bundle', type=str, metavar='BUNDLE_NAME',
+                        help='The name of the bundle')
+    parser.add_argument('--collection', type=str, metavar='COLLECTION_NAME',
+                        help='The name of the collection')
+    parser.add_argument('--collection-product-file', type=str,
+                        metavar='COLLECTION_PRODUCT_FILEPATH',
+                        help='The output location of the collection product.')
+
+    args = parser.parse_args()
+
+    collection_path = Path(args.collectionpath).absolute()
+    print(f"Creating collection product for: {collection_path}")
+    label_files = []
+
+    if args.bundle:
+        bundle = args.bundle
+    else:
+        bundle = collection_path.parent.name
+
+    if args.collection:
+        collection = args.collection
+    else:
+        collection = collection_path.name
+
+    print(f"Bundle name: {bundle}")
+    print(f"Collection name: {collection}")
+
+    primary = f"{bundle}:{collection}"
+
+    # Gather all XML files in the collection path, sort by directory structure
+    label_files = collection_path.glob('**/*.xml')
```

Comment:
As discussed, this needs to be `iglob` and move the if below to later.

Reply:
This uses pathlib.Path.glob, which returns a generator object, not a list of files.

Reply:
Oooo nice catch.